### ISSUE TITLE
False positive W0612 with dataclasses

### ISSUE THREAD

**Issue by DervishD** [microsoft/vscode-pylint#458]
With the minimal code shown below, I get error `W0612`,  unused variable `_HAS_DEFAULT_FACTORY` at line 1 column 1, the start of the code. The module `docstring` is not relevant, it is just there to show that the error is signalled at line 1, column 1, no matter the content.

Extension version is `v2023.10.1`, VSCode version is `1.84.2` on Windows 11 Home.

Using bundled pylint and my default configuration which shows all warnings (and I need to keep it that way).

```
"""."""

from dataclasses import dataclass

@dataclass
class SomeDataClass():
    """."""
```

This happens in larger code, in fact it triggered on a project I'm working on where I introduced a dataclass.

Feel free to contact me for further information needed or for any test I have to carry, and thanks a lot in advance.

**Comment by karthiknadig** [comment]
@DervishD Can you shar the logs from Output > Pylint? This might be a bug in pylint itself.

**Comment by DervishD** [comment]
@karthiknadig of course, I'm attaching it here as a text file. If you need more info just ask, and THANKS a lot for your interest. Sorry for the delay, today has been a bit messy.

As I said, I'm using the `pylint` bundled with the extension, but I can test with a locally installed one if needed.
[report.txt](https://github.com/microsoft/vscode-pylint/files/13429847/report.txt)

**Comment by karthiknadig** [comment]
This should be filed on pylint. All this extension does is parse and present the output in the problems view. Try running pylint the same version as the one in the extension, and see if it repros.

**Comment by DervishD** [comment]
Thanks @karthiknadig Tomorrow I will test with a locally installed version to see if the problem reproduces. If it does not, may it be due to some caching the pylint server is doing? Anyway, I don't want to advance events, so first I will carry some tests and if the problem reproduces I'll report directly there and I'll close this bug myself 😄

**Comment by DervishD** [comment]
Bug reported directly on `pylint`, after testing with a locally installed, up-to-date version, outside VS [https://github.com/pylint-dev/pylint/issues/9258](url)
